p2p/discover/v5_udp.go

```diff
@@ -832,17 +830,27 @@ func packNodes(reqid []byte, nodes []*enode.Node) []*v5wire.Nodes {
 		return []*v5wire.Nodes{{ReqID: reqid, Total: 1}}
 	}
 
-	total := uint8(math.Ceil(float64(len(nodes)) / 3))
+	// This limit represents the available space for nodes in output packets. Maximum
+	// packet size is 1280, and out of this ~80 bytes will be taken up by the packet
+	// frame. So limiting to 1000 bytes here leaves 200 bytes for other fields of the
+	// NODES message, which is a lot.
+	const sizeLimit = 1000
+
 	var resp []*v5wire.Nodes
 	for len(nodes) > 0 {
-		p := &v5wire.Nodes{ReqID: reqid, Total: total}
-		items := min(nodesResponseItemLimit, len(nodes))
-		for i := 0; i < items; i++ {
-			p.Nodes = append(p.Nodes, nodes[i].Record())
+		p := &v5wire.Nodes{ReqID: reqid}
+		size := uint64(0)
+		for len(nodes) > 0 && size < sizeLimit {
+			r := nodes[0].Record()
+			size += r.Size()
+			p.Nodes = append(p.Nodes, r)
+			nodes = nodes[1:]
```

Comment:
> So limiting to 1000 bytes here leaves 200 bytes for other fields

But it doesn't limit it to `1000` bytes, rather uses `1000` as the floor. So if it's at `999`, it might add another record of `3000` bytes, it may well blow over the limit. 

```suggestion
			if size + r.Size() < sizeLimit {
				p.Nodes = append(p.Nodes, r)
				nodes = nodes[1:]
				size += r.Size()
			}
```
Maybe something like that ?